Sources/MapboxMaps/Ornaments/Compass/MapboxCompassOrnamentView.swift

```diff
@@ -51,19 +51,25 @@ internal class MapboxCompassOrnamentView: UIButton {
                                               comment: "Accessibility hint")
 
         if let image = createCompassImage() {
-            containerView.image = image
-            NSLayoutConstraint.activate([
-                widthAnchor.constraint(equalToConstant: image.size.width),
-                heightAnchor.constraint(equalToConstant: image.size.height),
-                containerView.widthAnchor.constraint(equalToConstant: image.size.width),
-                containerView.heightAnchor.constraint(equalToConstant: image.size.height)
-            ])
+            updateImage(image: image)
         }
         containerView.translatesAutoresizingMaskIntoConstraints = false
         addSubview(containerView)
         addTarget(self, action: #selector(didTap), for: .touchUpInside)
     }
 
+    func updateImage(image: UIImage?) {
+        if image != nil {
+            containerView.image = image
+            NSLayoutConstraint.activate([
+                widthAnchor.constraint(equalToConstant: image!.size.width),
```

Comment:
Just want to make sure that this is intentional - previously the compass view were of size 40x40 pt, now we are giving the compass size up to the discretion of developers, is this how it behaves on Android? An alternative could be to keep the compass view size constant and scale the image up/down.

Reply:
>  now we are giving the compass size up to the discretion of developers, is this how it behaves on Android?

yes, to give context on how it works on Android, we expose:
 - gravity: allows to position an element (top-right by default for compass)
 - margins: allows to offset the view from the edge

>  An alternative could be to keep the compass view size constant and scale the image up/down.

That would limit the freedom of the developer to build any kind of UX. 
If we properly update the constraints, I'm expecting it should work for any image size.